### PR TITLE
[FE] [ADD] Eip 247 fe payment

### PR DESCRIPTION
**Pull request by wernerramon**
### 🛠 Changes being made

This PR adds the functionality to save payment methods to resto and visitor web. 
(The credit card component is mostly the same for both sides but the styling was not working for resto, thats why credit card components is in resto and in web)

Also be sure to add STRIPE_SECRET_KEY and STRIPE_PUBLISHABLE_KEY to the env file in the backend

### 🧪 Test plan

cypress tests

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/backend/cypress/e2e/payment.cy.ts
Comment (MarcPIster) [inline packages/backend/cypress/e2e/payment.cy.ts:1]: use single quotes for strings

File: packages/restoWeb/src/pages/Payments/PaymentsPage/PaymentPage.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                ) : (
+                    <div className={styles.noPaymentMethods}>
+                        {t('pages.Payment.nopay')}
+                    </div>
+                )}
+            </>
+            )}
+            <form className={styles.addButton} action="http://localhost:8081/api/payments/save/create-checkout-session" method="POST">
+                <input type="hidden" name="customerId" value={customerID} />
+                <input type="hidden" name="domainURL" value="http://localhost:8080" />
```
Comment (MarcPIster) [inline packages/restoWeb/src/pages/Payments/PaymentsPage/PaymentPage.tsx:81]: Shoudn't use localhost as static string.

File: packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx [hunk 2]
```diff
@@ -91,6 +91,7 @@ const MyAccountPage = () => {
 
   const fetchFavoriteDishes = async () => {
     const userToken = localStorage.getItem("user");
+    console.log(userToken);
```
Comment (MarcPIster) [inline packages/visitorWeb/src/pages/MyAccountPage/MyAccountPage.tsx:94]: remove console log

File: packages/visitorWeb/src/pages/Payments/PaymentsPage/PaymentPage.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                            onUpdate={fetchMethods}
+                        />
+                    ))}
+                </div>
+                ) : (
+                    <div className={styles.noPaymentMethods}>
+                        {t('pages.Payment.nopay')}
+                    </div>
+                )}
+                <form className={styles.addButton} action="http://localhost:8081/api/payments/save/create-checkout-session" method="POST">
```
Comment (MarcPIster) [inline packages/visitorWeb/src/pages/Payments/PaymentsPage/PaymentPage.tsx:77]: again, dont use localhost as static link

File: packages/visitorWeb/src/pages/Payments/PaymentsPage/PaymentPage.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    <input type="hidden" name="customerId" value={customerID} />
+                    <input type="hidden" name="domainURL" value="http://localhost:8082" />
```
Comment (MarcPIster) [inline packages/visitorWeb/src/pages/Payments/PaymentsPage/PaymentPage.tsx:79]: dont use localhost as static string in production

**Review by MarcPIster — COMMENTED** [review]
Please fix comments and double check if all localhost references are replaced

**Review by MarcPIster — APPROVED** [review]
